internal/tools/firestore/firestoreadddocuments/firestoreadddocuments.go

```diff
@@ -0,0 +1,220 @@
+	return kind
+}
+
+func (cfg Config) Initialize(srcs map[string]sources.Source) (tools.Tool, error) {
+	// verify source exists
+	rawS, ok := srcs[cfg.Source]
+	if !ok {
+		return nil, fmt.Errorf("no source named %q configured", cfg.Source)
+	}
+
+	// verify the source is compatible
+	s, ok := rawS.(compatibleSource)
+	if !ok {
+		return nil, fmt.Errorf("invalid source for %q tool: source kind must be one of %q", kind, compatibleSources)
+	}
+
+	// Create parameters
+	collectionPathParameter := tools.NewStringParameter(
+		collectionPathKey,
+		"The path of the collection where the document will be added to",
+	)
+
+	documentDataParameter := tools.NewMapParameter(
+		documentDataKey,
+		`The document data in Firestore's native JSON format. Each field must be wrapped with a type indicator:
+- Strings: {"stringValue": "text"}
+- Integers: {"integerValue": "123"} or {"integerValue": 123}
+- Doubles: {"doubleValue": 123.45}
+- Booleans: {"booleanValue": true}
+- Timestamps: {"timestampValue": "2025-01-07T10:00:00Z"}
+- GeoPoints: {"geoPointValue": {"latitude": 34.05, "longitude": -118.24}}
+- Arrays: {"arrayValue": {"values": [{"stringValue": "item1"}, {"integerValue": "2"}]}}
+- Maps: {"mapValue": {"fields": {"key1": {"stringValue": "value1"}, "key2": {"booleanValue": true}}}}
+- Null: {"nullValue": null}
+- Bytes: {"bytesValue": "base64EncodedString"}
+- References: {"referenceValue": "collection/document"}`,
+		"", // Empty string for generic map that accepts any value type
+	)
+
+	returnDocumentDataParameter := tools.NewBooleanParameterWithDefault(
+		returnDocumentDataKey,
+		false,
+		"If set to true the output will have the data of the created document. This flag if set to false will help avoid overloading the context of the agent.",
+	)
+
+	parameters := tools.Parameters{
+		collectionPathParameter,
+		documentDataParameter,
+		returnDocumentDataParameter,
+	}
+
+	mcpManifest := tools.McpManifest{
+		Name:        cfg.Name,
+		Description: cfg.Description,
+		InputSchema: parameters.McpManifest(),
+	}
+
+	// finish tool setup
+	t := Tool{
+		Name:         cfg.Name,
+		Kind:         kind,
+		Parameters:   parameters,
+		AuthRequired: cfg.AuthRequired,
+		Client:       s.FirestoreClient(),
+		manifest:     tools.Manifest{Description: cfg.Description, Parameters: parameters.Manifest(), AuthRequired: cfg.AuthRequired},
+		mcpManifest:  mcpManifest,
+	}
+	return t, nil
+}
+
+// validate interface
+var _ tools.Tool = Tool{}
+
+type Tool struct {
+	Name         string           `yaml:"name"`
+	Kind         string           `yaml:"kind"`
+	AuthRequired []string         `yaml:"authRequired"`
+	Parameters   tools.Parameters `yaml:"parameters"`
+
+	Client      *firestoreapi.Client
+	manifest    tools.Manifest
+	mcpManifest tools.McpManifest
+}
+
+func (t Tool) Invoke(ctx context.Context, params tools.ParamValues) (any, error) {
+	mapParams := params.AsMap()
+
+	// Get collection path
+	collectionPath, ok := mapParams[collectionPathKey].(string)
```

Comment:
not blocking: in the future we may want to improve validation of this path or improve the param description